sdk/core/azure-core/inc/azure/core/case_insensitive_map.hpp

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief A `map<string, string>` with case-insensitive key comparison.
+ */
+
+#pragma once
+
+#include "azure/core/internal/strings.hpp"
+
+#include <map>
+#include <string>
+
+namespace Azure { namespace Core {
+
+  /**
+   * @brief A type alias of `std::map<std::string, std::string>` with case-insensitive key
+   * comparison.
+   */
+  using CaseInsensitiveMap
+      = std::map<std::string, std::string, Internal::Strings::CaseInsensitiveComparator>;
```

Comment:
Can you please summarize why we decided against the "wrap this in our own type" approach that was discussed before?
https://github.com/Azure/azure-sdk-for-cpp/issues/1626#issuecomment-775353294

Reply:
This is simpler, we introduce less code, and it is familiar for user and less chance for us to introduce a bug. It satisfies all our requirements, metadata uses it, it is minimum invasive behavior. Jeffrey and Jinming liked it. Jinming still could make Metadata class into something else (storage uses, basically, a typedef), but I don't think he will, he liked the map.

Reply:
> Can you please summarize why we decided against the "wrap this in our own type" approach that was discussed before?

As I explained in this issue https://github.com/Azure/azure-sdk-for-cpp/issues/1640#issuecomment-776480022, doing so would be a lot of work, which I don't think is worth it.

I'm not saying you cannot do that, I'm just trying to say if you wanna do it, think thoroughly and do it right.